primitives/src/erasure_code/mod.rs

```diff
@@ -7,34 +7,19 @@
 //! Module for erasure code
 
 use crate::errors::PrimitivesError;
-use ark_ff::Field;
-use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
-use ark_std::{fmt::Debug, vec::Vec};
+use ark_std::vec::Vec;
 
 pub mod reed_solomon_erasure;
 
 /// Erasure code trait
-pub trait ErasureCode<F: Field>: Sized {
+/// `T` is the input data type
+pub trait ErasureCode<T>: Sized {
     /// Type for each data shards
-    type Shard: Debug
-        + Clone
-        + Eq
-        + PartialEq
-        + Sync
-        + Send
-        + CanonicalSerialize
-        + CanonicalDeserialize;
-
-    /// Create a new instance
-    ///  * `reconstruction_size`: and the minimum number of shards required for
-    ///    reconstruction
-    ///  * `num_shards`: the block (codeword) length
-    /// TODO should we allow constructors at the trait level?
-    fn new(reconstruction_size: usize, num_shards: usize) -> Result<Self, PrimitivesError>;
+    type Shard;
```

Comment:
I prefer to keep these trait bounds.

Reply:
As discussed offline, we'll keep the trait bounds due to issues discussed here:
- https://github.com/rust-lang/rust/issues/20671
- https://justin.restivo.me/posts/2022-10-16-dealing-with-unwieldy-generics.html